sdk-setup/src/mb2

```diff
@@ -1888,69 +1888,70 @@ sign_rpms_native() {
 
     local extra_args=()
 
-    if [[ $OPT_PACKAGE_SIGN_USER ]]; then
-
-        if inside_build_engine; then
-            local shared_key=${OPT_SHARED_DIR}/gnupg/${OPT_PACKAGE_SIGN_USER}.key
-            if [[ ! -f $shared_key ]]; then
-                notice "Cannot sign packages: Internal error: No GPG key shared for the user '$OPT_PACKAGE_SIGN_USER'."
-                return 1
-            fi
+    if [[ ! $OPT_PACKAGE_SIGN_USER ]]; then
+        fatal "Cannot sign packages: No signing user specified."
```

Comment:
It would be better to fail already at the time arguments are parsed, before the build starts. And not only in the case that signing user is not specified, but also in the other cases. Imagine the disappointment when build fails at the very end and it took 15 minutes to go there :)

So I would extract `sign_rpms_native__prepare` that would contain everything except the tmpdir creation and actual rpmsign call. There would be also `sign_rpms__prepare` which would do similar branching as the `sign_rpms` does now and this `sign_rpms__prepare` would be called from `run_build__process_args` and `run_package__process_args` if the `--sign` option is used.